SimPEG/electromagnetics/time_domain/simulation_1d_stitched.py

```diff
@@ -0,0 +1,208 @@
+        J = sim.getJ(model)
+        # we assumed the tx heights in a sounding is fixed
+        J["dh"] = J["dh"].sum(axis=1)
+        return J
+    else:
+        em_response = sim.dpred(model)
+        return em_response
+
+
+#######################################################################
+#       STITCHED 1D SIMULATION CLASS AND GLOBAL FUNCTIONS
+#######################################################################
+
+
+class Simulation1DLayeredStitched(BaseStitchedEM1DSimulation):
+    _simulation_type = "time"
+
+    def get_coefficients(self):
+        run_simulation = run_simulation_time_domain
+
+        if self.verbose:
+            print(">> Calculate coefficients")
+        if self.parallel:
+            pool = Pool(self.n_cpu)
+            self._coefficients = pool.map(
+                run_simulation,
+                [self.input_args_for_coeff(i) for i in range(self.n_sounding)],
+            )
+            self._coefficients_set = True
+            pool.close()
+            pool.join()
+        else:
+            self._coefficients = [
+                run_simulation(self.input_args_for_coeff(i))
+                for i in range(self.n_sounding)
+            ]
+
+    def forward(self, m):
+        self.model = m
+
+        if self.verbose:
+            print(">> Compute response")
+
+        # Set flat topo at zero
+        # if self.topo is None:
+
+        run_simulation = run_simulation_time_domain
+
+        if self.parallel:
+            if self.verbose:
+                print("parallel")
+            # This assumes the same # of layers for each of sounding
+            # if self.n_sounding_for_chunk is None:
+            pool = Pool(self.n_cpu)
+            result = pool.map(
+                run_simulation,
+                [
+                    self.input_args(i, output_type="forward")
+                    for i in range(self.n_sounding)
+                ],
+            )
+
+            pool.close()
+            pool.join()
+        else:
+            result = [
+                run_simulation(self.input_args(i, output_type="forward"))
+                for i in range(self.n_sounding)
+            ]
+
+        return np.hstack(result)
+
+    def getJ(self, m):
+        """
+        Compute d F / d sigma
+        """
+        self.model = m
+        if getattr(self, "_J", None) is None:
+            if self.verbose:
+                print(">> Compute J")
+
+            run_simulation = run_simulation_time_domain
+
+            if self.parallel:
+                if self.verbose:
+                    print(">> Start pooling")
+
+                pool = Pool(self.n_cpu)
+
+                # Deprecate this for now, but revisit later
+                # It is an idea of chunking for parallelization
+                # if self.n_sounding_for_chunk is None:
+                self._J = pool.map(
+                    run_simulation,
+                    [
+                        self.input_args(i, output_type="sensitivity")
```

Comment:
Looks like we are re-instantiating classes on every call? Can we not store the simulation?

Reply:
That's a good point. Yeah, I am instantiating classes on every call. I felt that this is more efficient than storing it and passes to child, but I could be wrong.